pkg/types/clusterconfig/clusterconfig.go

```diff
@@ -650,14 +657,47 @@ func (cc *Config) Validate(awsClient *aws.Client) error {
 	return nil
 }
 
-func CheckCortexSupport(instanceMetadata aws.InstanceMetadata) error {
+func checkCortexSupport(instanceMetadata aws.InstanceMetadata) error {
 	if strings.HasSuffix(instanceMetadata.Type, "nano") ||
 		strings.HasSuffix(instanceMetadata.Type, "micro") {
 		return ErrorInstanceTypeTooSmall()
 	}
 
-	if _, ok := awsutils.InstanceENIsAvailable[instanceMetadata.Type]; !ok {
-		return ErrorInstanceTypeNotSupported(instanceMetadata.Type)
+	if err := checkCNISupport(instanceMetadata.Type); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Check the instance type against the list of supported instance types for the current default CNI version
+// Returns nil if unable to perform the check successfully
+func checkCNISupport(instanceType string) error {
```

Comment:
This function overloads nil to represent a successful CNI support check and an error check. I think we should return nil only when the CNI support is successful. We should leave error handling to the caller. In this case the caller would swallow any error except for `ErrInstanceTypeNotSupported`.